test/e2e/llmisvc/test_llm_inference_service.py

```diff
@@ -82,35 +89,121 @@ class TestCase:
+                "fine-grained access control to data at rest and in motion, and identity management across disaggregated "
+                "components. Discuss how such an architecture impacts compliance with regulatory frameworks (e.g., GDPR, HIPAA) "
+                "concerning data sovereignty, privacy, and auditability. 5. Operational Complexity and Observability: "
+                "Examine the increased complexity in monitoring, logging, and tracing across highly decoupled compute and "
+                "data planes. What specialized tooling and practices (e.g., distributed tracing with OpenTelemetry, advanced AIOps) "
+                "would be essential? How would incident response and troubleshooting differ in this disaggregated environment "
+                "compared to traditional integrated systems? Consider the challenges of pinpointing root causes across "
+                "independent failures. 6. Real-world Applicability and Future Trends: Identify specific industries "
+                "or use cases (e.g., high-frequency trading, IoT edge processing, large language model inference) "
+                "where the benefits of P/D disaggregation would strongly outweigh its complexities. "
+                "Conclude by speculating on emerging technologies or paradigms (e.g., serverless compute functions "
+                "directly interacting with object storage, in-memory disaggregation) that could further drive or "
+                "transform P/D disaggregation in cloud computing.",
+                max_tokens=2000,
+            ),
+            marks=[
+                pytest.mark.cluster_gpu,
+                pytest.mark.cluster_nvidia,
+                pytest.mark.cluster_nvidia_roce,
+            ],
+        ),
+        pytest.param(
+            TestCase(
+                base_refs=[
+                    "router-no-scheduler",
+                    "workload-single-cpu",
+                    "model-fb-opt-125m",
+                ],
+                prompt="What is KServe?",
+            ),
+            marks=[
+                pytest.mark.cluster_cpu,
+                pytest.mark.cluster_single_node,
+                pytest.mark.no_scheduler,
+            ],
+        ),
+        pytest.param(
+            TestCase(
+                base_refs=[
+                    "router-managed",
+                    "workload-simulated-dp-ep-cpu",
+                    "model-fb-opt-125m",
+                ],
+                prompt="This test simulates DP+EP that can run on CPU, the idea is to test the LWS-based deployment, "
+                "but without the resources requirements for DP+EP (GPUs and ROCe/IB).",
+            ),
+            marks=[pytest.mark.cluster_cpu, pytest.mark.cluster_multi_node],
+        ),
     ],
     indirect=["test_case"],
     ids=generate_test_id,
 )
 @log_execution
 def test_llm_inference_service(test_case: TestCase):
+    inject_k8s_proxy()
```

Comment:
We should make it a setting for `KServeClient` as it might be needed for other tests. Can we plan a follow-up for this?

Reply:
I agree, I've tried but it seems an ugly hack to add to an SDK type of library

Reply:
and what I'm not unsure is what happens in-cluster only scenarios